File: Fuctions/financial_calculator_update.c
// Blaine Buckler, Financial Calculator C
#include <stdio.h>
    float income, rent, utilities, groceries, transportation, expenses, savings, total; 

void getUserInput(char type[], float *cash) {
    printf("How much %s? ", type);
    scanf("%f", cash);
    
}

void percent(char type[], int amount){
    int per = amount/ income *100; 

    printf("The percent of your income for %s is %d%%.\n", type, per); 
}

int main(void){
    printf("Hello and welcome to our budget calculator\n"); 
    getUserInput("do you make each month", &income);
    getUserInput("is your rent", &rent);
    getUserInput("are your utilities", &utilities);
    getUserInput("are your groceries?", &groceries);
    getUserInput("is your transportation?", &transportation);
    expenses = rent + utilities + groceries + transportation; 
    savings = income * .2; 
    total = income - savings - expenses; 
    printf("\nYour monthly income is $%.2f\n", income); 
    printf("\nYour monthly expenses is $%.2f\n", expenses); 
    printf("\nYour monthly savings is $%.2f\n", savings); 
    printf("\nYou have $%.2f left over\n", total); 
    percent("rent", rent); 
    percent("utilities", utilities); 
    percent("groceries", groceries); 
    percent("transportation", transportation); 
    percent("exspenses", expenses); 
    return 0;
}

File: Fuctions/financial_calculator_update.py
# Blaine Buckler, Financial Calculator Update Python

print("Welcome to my program, this program will calculate your finances. \n")


def get_input(prompt):  
    print("Please tell me your ", end="")
    return float(input(prompt))  
  
income = get_input("income: ")  
rent = get_input("rent: ")  
utilities = get_input("utilities: ")  
food = get_input("expenses: ")  
transport = get_input("costs: ")  


savings = income * 0.2
exspenses = rent + utilities + food + transport
spending = income - exspenses - savings

def percent(type, amount): 
    per = amount/income *100

    print(f"Your {type} is {per:.2f}% income. \n")

def cash(type, amount): 

    print(f"Your monthly {type} is ${amount:.2f}\n")

cash("rent", rent)
cash("exspenses", exspenses)
cash("savings", savings)
cash("spending", spending)

percent("rent", rent)
percent("utilities", utilities)
percent("groceries", food)
percent("transportation", transport)
percent("savings", savings)
percent("exspenses", exspenses)
